src/components/InputDetail.tsx

```diff
@@ -5,23 +5,19 @@ import Tab from "@mui/material/Tab";
 import TextField from "@mui/material/TextField";
 import { Stack } from "@mui/system";
 import { useTheme } from "@mui/material";
+import { AddTask } from "./model";
 
-interface Props {
-  addTask: string;
-  setAddTask: React.Dispatch<React.SetStateAction<string>>;
-  handleAdd: (e: React.FormEvent) => void;
-}
 const frequencies = ["daily", "weekly", "monthly", "yearly"];
 
-const InputDetail: React.FC<Props> = ({ addTask, setAddTask, handleAdd }) => {
+const InputDetail: React.FC<AddTask> = ({ addTask, setAddTask, handleAdd }) => {
```

Comment:
you'll need to destructure out the setter functions you passed in, so you can use them in the onchanges for each input

```suggestion
const InputDetail: React.FC<AddTask> = ({ setTaskTitle, setTaskDescription, setTaskFrequency }) => {
```